nativelib/src/main/scala/scala/scalanative/runtime/SocketHelpers.scala

```diff
@@ -0,0 +1,173 @@
+package scala.scalanative.runtime
+
+import scalanative.native._
+import scalanative.posix.{netdb, netdbOps}, netdb._, netdbOps._
+import scalanative.posix.arpa.inet._
+import scalanative.posix.sys.socketOps._
+import scalanative.posix.sys.socket._
+import scalanative.posix.sys.select._
+import scalanative.posix.unistd.close
+import scalanative.posix.sys.selectOps._
+import scalanative.posix.netinet.{in, inOps}, in._, inOps._
+
+object SocketHelpers {
+
+  def isReachableByEcho(ip: String, timeout: Int): Boolean = {
```

Comment:
Given that this is the only function that is actually used, I think that you should put this function directly in `InetAddress`, make it `private[net]`, add a parameter to define the port and write a scripted test for it.